### ISSUE TITLE
pix2pix for Rgb to thermal image translation

### ISSUE THREAD

**Issue by durgesh17media** [junyanz/pytorch-CycleGAN-and-pix2pix#1037]
has anyone performed pix2pix approach for RGB to Thermal image translation?
I am getting very bad results while testing. anyone can help? 
 I used this command for testing:
python3 test.py --dataroot ./datasets/combine_faces/ --name face4_pix2pix --model pix2pix --direction AtoB

results: 
**Real A**

![1100_real_A](https://user-images.githubusercontent.com/65701365/82530185-63af0d80-9b5a-11ea-91a2-9fe9e420c0f9.png)

**Real B**
![1100_real_B](https://user-images.githubusercontent.com/65701365/82530188-6447a400-9b5a-11ea-979d-f40530845e28.png)

**FakeB**
![1100_fake_B](https://user-images.githubusercontent.com/65701365/82530180-61e54a00-9b5a-11ea-971d-b6d0505124d1.png)

**Comment by durgesh17media** [comment]
> has anyone performed pix2pix approach for RGB to Thermal image translation?
> I am getting very bad results while testing. anyone can help?
> I used this command for testing:
> python3 test.py --dataroot ./datasets/combine_faces/ --name face4_pix2pix --model pix2pix --direction AtoB
> 
> results:
> **Real A**
> 
> ![1100_real_A](https://user-images.githubusercontent.com/65701365/82530185-63af0d80-9b5a-11ea-91a2-9fe9e420c0f9.png)
> 
> **Real B**
> ![1100_real_B](https://user-images.githubusercontent.com/65701365/82530188-6447a400-9b5a-11ea-979d-f40530845e28.png)
> 
> **FakeB**
> ![1100_fake_B](https://user-images.githubusercontent.com/65701365/82530180-61e54a00-9b5a-11ea-971d-b6d0505124d1.png)

**Comment by junyanz** [comment]
If you have many training images, you can try [pix2pixHD](https://github.com/NVIDIA/pix2pixHD) or [SPADE](https://github.com/NVlabs/SPADE).

**Comment by durgesh17media** [comment]
thanks for the reply, I have 1400 rgb and thermal image pair. will it be ok for pix2pix hd or spade?

**Comment by junyanz** [comment]
You can use the default parameters. I also recommend that you use random cropping (e.g., `--load_size 143 --crop_size 128` and flipping.

**Comment by durgesh17media** [comment]
This you are telling for pix2pixHD?

**Comment by junyanz** [comment]
You can try both. Pix2pixHD might have different naming conversions for these flags.

**Comment by durgesh17media** [comment]
Still getting bad results after --load_size 143 --crop_size 128 .
Training command :python3 train.py --dataroot ./datasets/combine_faces/ --name face_new12_pix2pix --model pix2pix --direction AtoB --netG unet_128 --n_epochs 200 --n_epochs_decay 200 --preprocess none --load_size 143 --crop_size 128

![1004_fake_B](https://user-images.githubusercontent.com/65701365/82981287-52a64680-a009-11ea-88c5-e762671507ff.png)
![1004_real_A](https://user-images.githubusercontent.com/65701365/82981290-53d77380-a009-11ea-957e-14f57a0e3a4d.png)
![1004_real_B](https://user-images.githubusercontent.com/65701365/82981292-54700a00-a009-11ea-89fd-1046c37b702a.png)
![1005_fake_B](https://user-images.githubusercontent.com/65701365/82981293-5508a080-a009-11ea-99e8-6435979ce765.png)
![1005_real_A](https://user-images.githubusercontent.com/65701365/82981295-5508a080-a009-11ea-85e5-f59ba035f745.png)
![1005_real_B](https://user-images.githubusercontent.com/65701365/82981296-55a13700-a009-11ea-86e8-1918558597f3.png)
![1001_fake_B](https://user-images.githubusercontent.com/65701365/82981297-5639cd80-a009-11ea-907d-d2501a0b1885.png)
![1001_real_A](https://user-images.githubusercontent.com/65701365/82981299-5639cd80-a009-11ea-809c-aa85f179f8a6.png)
![1001_real_B](https://user-images.githubusercontent.com/65701365/82981301-56d26400-a009-11ea-957e-199e00fd3977.png)
![1002_fake_B](https://user-images.githubusercontent.com/65701365/82981302-576afa80-a009-11ea-9710-c960579b882b.png)
![1002_real_A](https://user-images.githubusercontent.com/65701365/82981303-58039100-a009-11ea-853e-229086b6ca05.png)
![1002_real_B](https://user-images.githubusercontent.com/65701365/82981305-58039100-a009-11ea-8123-5b7677ee3cad.png)
![1003_fake_B](https://user-images.githubusercontent.com/65701365/82981306-589c2780-a009-11ea-9812-38e717a47d29.png)
![1003_real_A](https://user-images.githubusercontent.com/65701365/82981309-5934be00-a009-11ea-879d-8e5a3026be79.png)
![1003_real_B](https://user-images.githubusercontent.com/65701365/82981310-5934be00-a009-11ea-924a-fa7717d48531.png)
 @junyanz please guide.

**Comment by junyanz** [comment]
1. Not sure how to further improve the quality. Maybe you can try different generators such as `--netG resnet_6blocks` or `--netG resnet_9blocks`. But there is no guarantee. 
2. 1. If your training data only contains faces of one person or a few persons, it may not work for your face. 2. Also for your new test data, you need to align and crop it before feeding to the network. 3. For a single test image, you need to use `--model test`. See this [script](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/scripts/test_single.sh) for a reference.

**Comment by durgesh17media** [comment]
hello @junyanz , can you tell me what is this **.ipynb_checkpoints** folder in pytorch-CycleGAN-and-pix2pix/results/face_new6_pix2pix/test_latest/images???? what it is being created and what is the use of this?

**Comment by durgesh17media** [comment]
@junyanz please respond on the above question.

**Comment by junyanz** [comment]
This [post](https://stackoverflow.com/questions/46421663/what-are-jupyter-notebook-checkpoint-files-for?rq=1) explains the ipytnb_checkpoints. It might be caused by the creation of a python notebook.

**Comment by sobhani** [comment]
Do you think if we play with the learning rate to higher (?) or lower(?), might affects the performance of the networks having better-generated results? else what could help for generating better outcomes? either in oix2pix or Pix2pixHD?

**Comment by junyanz** [comment]
Not sure. It all depends on your application and datasets. Quite hard to predict which learning rate is the best in advance.
